r/src/RecordBatch.cpp

```diff
@@ -96,23 +97,31 @@ int RecordBatch__to_file(const std::shared_ptr<arrow::RecordBatch>& batch,
 }
 
 // [[Rcpp::export]]
-RawVector RecordBatch__to_raw(const std::shared_ptr<arrow::RecordBatch>& batch) {
-  MemoryPool* pool = default_memory_pool();
+RawVector RecordBatch__to_stream(const std::shared_ptr<arrow::RecordBatch>& batch) {
+  std::unique_ptr<io::MockOutputStream> mockSink;
+  mockSink.reset(new io::MockOutputStream());
 
-  std::shared_ptr<ResizableBuffer> buffer;
-  R_ERROR_NOT_OK(AllocateResizableBuffer(pool, 0, &buffer));
+  std::shared_ptr<arrow::ipc::RecordBatchWriter> mockWriter;
+  R_ERROR_NOT_OK(arrow::ipc::RecordBatchStreamWriter::Open(mockSink.get(),
+                                                           batch->schema(),
+                                                           &mockWriter));
 
-  std::unique_ptr<io::BufferOutputStream> sink;
-  sink.reset(new io::BufferOutputStream(buffer));
+  R_ERROR_NOT_OK(mockWriter->WriteRecordBatch(*batch));
+  R_ERROR_NOT_OK(mockWriter->Close());
+
+  RawVector res(mockSink->GetExtentBytesWritten());
+
+  std::unique_ptr<RawVectorOutputStream> sink;
+  sink.reset(new RawVectorOutputStream(res));
```

Comment:
This seems a bit elaborate. Is there a way to get the pointer to the raw memory in `res`? Then you can just use `FixedSizeBufferWriter` https://github.com/apache/arrow/blob/master/cpp/src/arrow/io/memory.h#L103 cc @romainfrancois

Reply:
`res.begin()`

Reply:
OK, then `RawVectorOutputStream` is not needed